troposphere/emr.py

```diff
@@ -0,0 +1,186 @@
+            raise ValueError("Configuration '%s' must be of "
+                             "Configuration type" % x)
+    return xs
+
+
+class Configuration(AWSProperty):
+    props = {
+        'Classification': (basestring, False),
+        'ConfigurationProperties': (properties_validator, False),
+        'Configurations': (configurations_validator, False)
+    }
+
+
+def market_validator(x):
+    valid_values = ['ON_DEMAND', 'SPOT']
+    if x not in valid_values:
+        raise ValueError("Market must be one of: %s" %
+                         ', '.join(valid_values))
+    return x
+
+
+class InstanceGroupConfigProperty(AWSProperty):
+    props = {
+        'BidPrice': (basestring, False),
+        'Configurations': (configurations_validator, False),
+        'InstanceCount': (integer, True),
+        'InstanceType': (basestring, True),
+        'Market': (market_validator, False),
+        'Name': (basestring, False)
+    }
+
+
+class PlacementType(AWSProperty):
+    props = {
+        'AvailabilityZone': (basestring, True)
+    }
+
+
+class JobFlowInstancesConfig(AWSProperty):
+    props = {
+        'AdditionalMasterSecurityGroups': (list, False),
+        'AdditionalSlaveSecurityGroups': (list, False),
+        'CoreInstanceGroup': (InstanceGroupConfigProperty, True),
+        'Ec2KeyName': (basestring, False),
+        'Ec2SubnetId': (basestring, False),
+        'EmrManagedMasterSecurityGroup': (basestring, False),
+        'EmrManagedSlaveSecurityGroup': (basestring, False),
+        'HadoopVersion': (basestring, False),
+        'MasterInstanceGroup': (InstanceGroupConfigProperty, True),
+        'Placement': (PlacementType, False),
+        'ServiceAccessSecurityGroup': (basestring, False),
+        'TerminationProtected': (boolean, False)
+    }
+
+
+class Cluster(AWSObject):
+    resource_type = "AWS::EMR::Cluster"
+
+    props = {
+        'AdditionalInfo': (dict, False),
+        'Applications': ([Application], False),
+        'BootstrapActions': ([BootstrapActionConfig], False),
+        'Configurations': (configurations_validator, False),
+        'Instances': (JobFlowInstancesConfig, True),
+        'JobFlowRole': (basestring, True),
+        'LogUri': (basestring, False),
+        'Name': (basestring, True),
+        'ReleaseLabel': (basestring, False),
+        'ServiceRole': (basestring, True),
+        'Tags': (list, False),
+        'VisibleToAllUsers': (boolean, False)
+    }
+
+
+class InstanceGroupConfig(AWSObject):
+    resource_type = "AWS::EMR::InstanceGroupConfig"
+
+    props = {
+        'BidPrice': (basestring, False),
+        'Configurations': (configurations_validator, False),
+        'InstanceCount': (integer, True),
+        'InstanceRole': (basestring, True),
+        'InstanceType': (basestring, True),
+        'Market': (market_validator, False),
+        'Name': (basestring, False)
+    }
+
+
+class HadoopJarStepConfig(AWSProperty):
+    props = {
+        'Args': ([basestring], False),
```

Comment:
Yeah, there are some inconsistencies in the documentation, seems they wrote it in a rush! It seems that the actual value is a list of strings, as the doc example points out.

Will double check though.